docs/configure-rbac-permissions.md

```diff
@@ -0,0 +1,146 @@
+
+To help configure it successfully, in this post, we will start by describe a method to correctly configure the permissions for an RBAC enabled cluster. And then, we will share a guidance for troubleshooting.
+
+## Configure ClusterRole and ClusterRoleBinding for the service account
+
+* Create a yaml file, `sa-role.yaml` for example. We will deploy it when it is ready.
+
+* Write spec to define a cluster role, name it `appinsights-k8s-property-reader` for example:
+ 
+    ```yaml
+    kind: ClusterRole
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+    # "namespace" omitted since ClusterRoles are not namespaced
+    name: appinsights-k8s-property-reader
+    rules:
+    - apiGroups: ["", "apps"]
+    resources: ["pods", "nodes", "replicasets", "deployments"]
+    verbs: ["get", "list"]
+    ```
+    That spec defines the name of the role, and what permission does the role has, for example, list pods.
+
+    You don't have to use the exact name, but you will need to making sure the name is referenced correctly in the following steps.
+
+* Append a Cluster role binding spec, you will need to update the namespace accordingly:
+
+    ```yaml
+    ---
+    kind: ClusterRoleBinding
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+    name: appinsights-k8s-property-reader-binding
+    subjects:
+    - kind: ServiceAccount
+    name: default
+    namespace: default
+    roleRef:
+    kind: ClusterRole
+    name: appinsights-k8s-property-reader
+    apiGroup: rbac.authorization.k8s.io
+    ```
+    
+    That is to grant the role of `appinsights-k8s-property-reader` to the default service account.
+    
+* Now you can deploy it:
+
+    ```shell
+    kubectl create -f `sa-role.yaml`
+    ```
+    See [sa-role.yaml](sa-role.yaml) for a full example.
+
+> :warning: Check back for various permissions needed. Depends on the implementations, it may change in the over time.
+
+## Ad-hoc troubleshooting for permission
+
+Kubectl provides an `auth --can-i` sub command for troubleshooting permissions. It supports impersonate the service account. We can leverage it for permission troubleshooting, for example:
+
+```shell
+kubectl auth can-i list pod --namespace default --as system:serviceaccount:default:default
+```
+
+There are several `default` there, let's break it down:
+
+* --namespace **default**: This is the target namespace, it is usually where your applications deploys to.
+* system:serviceaccount:**default**:**default**: The first default is the **namespace** of the service account. The second is the **name** of your service account.
+
+The response for the command-line will simply be `yes` or `no`, for example:
+
+```shell
+$ kubectl auth can-i list pod --namespace default --as system:serviceaccount:default:default
+yes
+```
+
+## Use SubjectAccessReview
+
+Kubernetes also provides `SubjectAccessReview` to check permission for given user on target resource.
```

Comment:
Side note: did not know about `SubjectAccessReview`... I am not sure I like the pattern that this CRD uses, namely, "create an object and then wait for the controller to fill its status" as a way to make a request against the server. I think I would prefer an API aggregation approach instead. Oh well 🤷‍♂️

Reply:
That was a #TIL for me too. One details though, when I try to query the objects after I created them, the query failed. I don't know if whether those objects really persistent there or are they just `temporary` to hold the status and return and immediately got disposed.

I like it b/c it give me a way to share a yaml for others to use.